pallets/fragments/src/lib.rs

```diff
@@ -1383,45 +1384,63 @@ pub mod pallet {
 		/// # Arguments
 		///
 		/// * `origin` - The origin of the extrinsic function
-		/// * `definition_hash` - **ID of the Fragment Definition** to **detach**
+		/// * `definition_hash` - **ID of the Fragment Definition**
+		/// * `edition_ids` - **Edition IDs** of the **Fragment Instances that will be detached**
 		/// * `target_chain` - **External Blockchain** to attach the Proto-Fragment into
 		/// * `target_account` - **Public Account Address in the External Blockchain `target_chain`**
 		///   to assign ownership of the Proto-Fragment to
 		#[pallet::weight(25_000)] // TODO - weight
 		pub fn detach(
 			origin: OriginFor<T>,
 			definition_hash: Hash128,
-			edition_id: InstanceUnit,
-			copy_id: InstanceUnit,
+			edition_ids: Vec<InstanceUnit>,
 			target_chain: SupportedChains,
 			target_account: BoundedVec<u8, T::DetachAccountLimit>, // an eth address or so
 		) -> DispatchResult {
-
 			let who = ensure_signed(origin)?;
 
+			ensure!(!edition_ids.is_empty(), Error::<T>::InstancesToDetachIsEmpty);
 			ensure!(!target_account.is_empty(), Error::<T>::DetachAccountIsEmpty);
 
-			let current_block_number = <frame_system::Pallet<T>>::block_number();
+			// let current_block_number = <frame_system::Pallet<T>>::block_number();
+
+			let definition_permissions =
+				<Definitions<T>>::get(definition_hash).ok_or(Error::<T>::NotFound)?.permissions;
 
-			// owner must own instance
-			let owned_instances = <Inventory<T>>::get(who, definition_hash).ok_or(Error::<T>::NoPermission)?;
 			ensure!(
-				owned_instances.contains(&(Compact(edition_id), Compact(copy_id))),
+				(definition_permissions & FragmentPerms::COPY) != FragmentPerms::COPY,
```

Comment:
btw also COPIABLE items cannot be detached I guess for now

Reply:
Yes. That's why I check if the Fragment Definition is copyable - rather than checking if the item is copyable

Reply:
> btw also COPIABLE items cannot be detached I guess for now

Hmm but some copyable items can be `give()`en and its copy permission removed 🤔. That's true.

**So should I also accept Copy ID(s) as a parameter? ** Cos in the spec you previously mentioned, you only wanted to accept `edition_ids` @sinkingsugar

Reply:
Did we clarify this on discord @aang114 ?